display-layer-view-state/src/main/java/com/esri/arcgisruntime/sample/displaylayerviewstate/MainActivity.java

```diff
@@ -0,0 +1,153 @@
+
+    private static final int MIN_SCALE = 40000000;
+    private static final int TILED_LAYER = 0;
+    private static final int IMAGE_LAYER = 1;
+    private static final int FEATURE_LAYER = 2;
+    private MapView mMapView;
+    private TextView timeZoneTextView;
+    private TextView worldCensusTextView;
+    private TextView recreationTextView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        // create three layers to add to the map
+        final ArcGISTiledLayer tiledLayer = new ArcGISTiledLayer(getApplication().getString(R.string.world_timezone_service_URL));
+
+        final ArcGISMapImageLayer imageLayer = new ArcGISMapImageLayer(getApplication().getString(R.string.world_census_service_URL));
+        // setting the scales at which this layer can be viewed
+        imageLayer.setMinScale(MIN_SCALE);
+        imageLayer.setMaxScale(MIN_SCALE / 10);
+
+        // creating a layer from a service feature table
+        final ServiceFeatureTable featureTable = new ServiceFeatureTable(getApplication().getString(R.string.world_facilities_service_URL));
+        final FeatureLayer featureLayer = new FeatureLayer(featureTable);
+
+        // inflate MapView from layout
+        mMapView = (MapView) findViewById(R.id.mapView);
+        // create a map with the BasemapType topographic
+        final ArcGISMap mMap = new ArcGISMap(Basemap.createTopographic());
+        // add the layers on the map
+        mMap.getOperationalLayers().add(tiledLayer);
+        mMap.getOperationalLayers().add(imageLayer);
+        mMap.getOperationalLayers().add(featureLayer);
+
+        // set the map to be displayed in this view
+        mMapView.setMap(mMap);
+
+        // inflate TextViews from the layout
+        timeZoneTextView = (TextView) findViewById(R.id.worldTimeZoneStatusView);
+        worldCensusTextView = (TextView) findViewById(R.id.censusStatusView);
+        recreationTextView = (TextView) findViewById(R.id.facilitiesStatusView);
+
+        // zoom to custom ViewPoint
+        mMapView.setViewpoint(new Viewpoint(
+                new Point(-11e6, 45e5, SpatialReferences.getWebMercator()), MIN_SCALE));
+
+        // Listen to changes in the status of the Layer
+        mMapView.addLayerViewStateChangedListener(new LayerViewStateChangedListener() {
+            @Override
+            public void layerViewStateChanged(LayerViewStateChangedEvent layerViewStateChangedEvent) {
+
+                // get the layer which changed it's state
+                Layer layer = layerViewStateChangedEvent.getLayer();
+
+                // get the View Status of the layer
+                // View status will be either of ACTIVE, ERROR, LOADING, NOT_VISIBLE, OUT_OF_SCALE, UNKNOWN
+                String viewStatus = layerViewStateChangedEvent.getLayerViewStatus().iterator().next().toString();
```

Comment:
This line is not getting invoked when the World Time Zones layer changes resulting in a `viewStatus=LOADED` on line 103 even when it is not loaded.  Reference the screenshot below to show the issue in the app:

![time-zone-active](https://cloud.githubusercontent.com/assets/1308001/16364517/e0266aaa-3b9d-11e6-9669-e97fccc01106.png)